src/Microsoft.AspNetCore.Hosting/Internal/HostingApplication.cs

```diff
@@ -64,21 +75,68 @@ public Context CreateContext(IFeatureCollection contextFeatures)
                 // Non-inline
                 LogRequestStarting(httpContext);
             }
-
             if (diagnoticsEnabled)
             {
+#if !NET451
+                context.Activity = StartActivity(httpContext);
+#endif
                 // Non-inline
                 RecordBeginRequestDiagnostics(httpContext, startTimestamp);
             }
 
-            // Create and return the request Context
-            return new Context
+            // Fill and return the request Context
+            context.HttpContext = httpContext;
+            context.Scope = scope;
+            context.StartTimestamp = startTimestamp;
+            return context;
+        }
+
+#if !NET451
+        private Activity StartActivity(HttpContext httpContext)
+        {
+            var activity = new Activity(ActivityName);
+            StringValues requestId;
+            if (httpContext.Request.Headers.TryGetValue(_activityTrackingOptions.RequestIdHeaderName, out requestId))
+            {
+                try
+                {
+                    activity.SetParentId(requestId);
+                }
+                catch (ArgumentException ex)
```

Comment:
If you are going to try to swallow exceptions you should catch them all, which means the  type the catch with 'Exception' rather than ArgumentException (This issue is in multiple places).    I would also prefer a try-catch around most of the routine (simpler).   I realize this gives slightly poorer error messages, but frankly the likelihood of these errors even happening, let alone the more detailed error message being useful is very low.